### PR TITLE
Database stuff

### PR DESCRIPTION
**Pull request by zsarvas**
added database functionality

### CODE REVIEW

File: src/application/delegator.go [hunk 1]
```diff
@@ -94,6 +94,8 @@ func (d *Delegator) HandleIncomingCommand() {
 // If no player exists, makes a new one and returns it
 func (d Delegator) fetchPlayer() domain.Player {
 	incomingId := d.DiscordUser.Author.String()
+
+	//fmt.Printf("fixed id: '%v'", fixedId)
```
Comment (RitterGustave-TR) [inline src/application/delegator.go]: Don't need this comment

File: src/infrastructure/playerHandler.go [hunk 2]
```diff
@@ -1,38 +1,126 @@
...
+	_ "github.com/mattn/go-sqlite3"
 	"github.com/zsarvas/RL-Discord-Matchmaking/domain"
 )
 
 type PlayerHandler struct {
-	players map[string]domain.Player
+	Conn *sql.DB
+}
+
+type Row interface {
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: Do we really use this interface anywhere?

File: src/infrastructure/playerHandler.go [hunk 3]
```diff
@@ -1,38 +1,126 @@
...
+	Scan(dest ...interface{})
+	Next() bool
 }
 
 func (handler *PlayerHandler) Add(newPlayer domain.Player) {
-	handler.players[newPlayer.Id] = newPlayer
+	handler.Conn.Exec(fmt.Sprintf(`INSERT INTO players (UID, Name, MMR, Wins, Losses) VALUES ('%v', '%v', '%f', '%d', '%d')`, newPlayer.Id, newPlayer.DisplayName, newPlayer.Mmr, newPlayer.NumWins, newPlayer.NumLosses))
 }
 
 func (handler *PlayerHandler) Remove(id string) {
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: We discussed this but we might not need this function

File: src/infrastructure/playerHandler.go [hunk 4]
```diff
@@ -1,38 +1,126 @@
...
+	panic(1)
 }
 
 func (handler *PlayerHandler) GetById(id string) domain.Player {
-	player, playerExists := handler.players[id]
 
-	if !playerExists {
-		newPlayerToAdd := domain.NewPlayer(id)
-		handler.Add(*newPlayerToAdd)
+	fmt.Printf("id is : '%v' \n", id)
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: we don't need this log

File: src/infrastructure/playerHandler.go [hunk 5]
```diff
@@ -1,38 +1,126 @@
...
+	pingError := connection.Ping()
+
+	if pingError != nil {
+		log.Fatal(pingError)
 	}
 
+	plyrHandler := &PlayerHandler{Conn: connection}
+
+	createTable(connection)
+	//showTable(connection)
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: Don't need this comment

File: src/infrastructure/playerHandler.go [hunk 6]
```diff
@@ -1,38 +1,126 @@
...
+
+func createTable(db *sql.DB) {
+
+	players_table := `CREATE TABLE IF NOT EXISTS players (
+		id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+		"UID" TEXT,
+		"Name" TEXT,
+		"MMR" INT,
+		"Wins" INT,
+		"Losses" INT);`
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: we should find a way to remove UID

File: src/infrastructure/playerHandler.go [hunk 7]
```diff
@@ -1,38 +1,126 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+		fmt.Println(uid)
+		fmt.Println(username)
+		fmt.Println(mmr)
+		fmt.Println(wins)
+		fmt.Println(losses)
+	}
+}*/
```
Comment (RitterGustave-TR) [inline src/infrastructure/playerHandler.go]: Unneeded comments perhaps

**Review by RitterGustave-TR — CHANGES_REQUESTED** [review]

**Comment by zsarvas** [comment]
requested changes have been made

**Pull request closed without merging**